text/0000-guides-restructure.md

```diff
@@ -0,0 +1,397 @@
+
+#### Templates & Template helpers
+
+Templates come before components, since it is possible for someone to not know JavaScript/Handlebars and still contribute to Ember. For example, designers could effectively work in only templates, using plain HTML. To make Templates work as the opening topic, additions are necessary in the introduction.
+
+Template Syntax will cover Handlebars basics and built-in features like `{{input}}`, `{{#each}}`, and `{{#if}}`.
+
+Template helpers are split out and moved to "Leveling Up" because developers do not need to use them to write basic Ember apps (they are not a prerequisite). This section includes writing custom Helpers and using built-in helpers like `get`, `let`, and `array`. Although helpers themselves are not considered a difficult topic, they are used most often when developers understand their app's architecture and the flow of data. Also, template helpers have more in common with JavaScript functions than concerns relating to layout and data flow.
+
+Lastly, another reason to split out Template syntax from Template Helpers is because the syntax section is expected to expand with the addition of Octane content.
+
+An alternative to this split is to reduce the documentation of helpers themselves in the Guides, and lean on the API docs instead, however great care should be taken in removing documentation. Without a strong case for it, we lean towards leaving them in the Guides.
+
+#### Working with JavaScript
+
+A common concern of new learners is, what is JavaScript and what is special to Ember? The Working with JavaScript section incorporates the existing JavaScript Primer and new content that will help make the distinctions clearer. Ember.js has procatively adopted JavaScript APIs as they become available, such as Classes and Decorators, in some cases before the larger JavaScript community becomes familiar with them. Key people in the Ember Community are also involved in the development of JavaScript itself through TC39. To an extent, it is our responsibility to at least suggest to developers which JavaScript concepts they must learn in order to feel comfortable with Ember.
+
+#### Components
+
+Components will need the most refactor work and new content for Octane, so here’s the subtopic breakdown:
+
+- Creating a component (includes Component Types post-Octane)
+- Displaying data in a Template
+- Adding actions
+- Nesting components (includes suggestions of when to break things into components)
+- Passing arguments
+- Using computed properties (extremely basic example, link to the Computed Property section)
+- Working with arrays of data
+- Lifecycle hooks
```

Comment:
I think the outline for this section needs to be revisted again in terms of general patterns, e.g. adding actions, passing arguments or state management with computeds/tracked vs. component-type related topics. There is ember-components, sparkles/glimmer components, hooked components, template only components ... and all of them have different APIs (template only has none! :D) but lifecycle hooks are different for those types of components. The guides shall not spread the idea of ambiguity amongst them.

Reply:
Technically it would make sense to start with embers component manager and explain there can be different types of components ... and this will be the point where the guides will state ember as suuuper difficult. Not a good way to get learners onboarded but technically not wrong :D

Instead, what may be more helpful to pick a little example that states:

- how to create a component
- how attributes and arguments work (and how to handle defaults for `@arguments` - this WILL come up)
- how to bring properties from component to template (in combination with state managemet)
- how modifiers interact with components at the example of action
- ..... where we'll come to lifecycle hooks stating different types of components

From your list "Displaying data in a Template" and "Working with arrays of data" seems like repetitive (only given the headlines). Nesting components is an advanced topic, which would better interact with given patterns (not only for ember) that exist for components and how to apply them with ember, which opens another section for the guides (what I just send you over at discord).

The list I outlined might still be technically driven but I'm quite happy with my grouping now make it approachable to beginners =)